src/common/recordbatch/src/adapter.rs

```diff
@@ -365,8 +366,7 @@ impl ExecutionPlanVisitor for MetricCollector {
     }
 
     fn post_visit(&mut self, _plan: &dyn ExecutionPlan) -> std::result::Result<bool, Self::Error> {
-        // the last minus will underflow
-        self.current_level = self.current_level.wrapping_sub(1);
```

Comment:
LGTM, but why change this line?

Reply:
> LGTM, but why change this line?

The potention underflow here is caused by a missing `self.current_level += 1`. After this fix there should be no underflow anymore.